metricflow-semantics/metricflow_semantics/specs/metadata_spec.py

```diff
@@ -17,9 +19,16 @@ class MetadataSpec(InstanceSpec):
     element_name: str
     agg_type: Optional[AggregationType] = None
 
-    @property
-    def qualified_name(self) -> str:  # noqa: D102
-        return f"{self.element_name}{DUNDER}{self.agg_type.value}" if self.agg_type else self.element_name
+    @override
+    @cached_property
+    def structured_name(self) -> StructuredLinkableSpecName:
+        if self.agg_type is not None:
+            return StructuredLinkableSpecName(
+                entity_link_names=(self.element_name,),
+                element_name=self.agg_type.value,
+            )
```

Comment:
If there is an `agg_type` present here, then you'll end up with an incorrect `element_name`. If we need this attribute on this class, we should add an `agg_type` attribute to `StructuredLinkableSpecName` and handle the dundering in the `qualified_name` attribute on that class.